scriptmodules/helpers.sh

```diff
@@ -345,13 +347,19 @@ function gitPullOrClone() {
         popd > /dev/null
     else
         local git="git clone --recursive"
-        if [[ "$__persistent_repos" -ne 1 && "$repo" == *github* ]]; then
+        if [[ "$__persistent_repos" -ne 1 && "$repo" == *github* && -z "$commit" ]]; then
             git+=" --depth 1"
         fi
         [[ "$branch" != "master" ]] && git+=" --branch $branch"
         echo "$git \"$repo\" \"$dir\""
         runCmd $git "$repo" "$dir"
     fi
+    if [[ "$commit" ]]; then
+        pushd "$dir" > /dev/null
+        echo "Winding back $repo: $branch to commit #$commit"
+        runCmd git checkout $commit
```

Comment:
is it possible to use `git -C` rather than pushd / popd ?

Reply:
`-C`.  Every day's a schools day.  Should work fine.  Building now.